zfs_test.go

```diff
@@ -1,422 +1,343 @@
 package zfs_test
 
 import (
-	"fmt"
 	"io/ioutil"
-	"math"
 	"os"
 	"path/filepath"
-	"reflect"
 	"runtime"
 	"testing"
-	"time"
 
 	zfs "github.com/mistifyio/go-zfs/v3"
 )
 
-func sleep(delay int) {
-	time.Sleep(time.Duration(delay) * time.Second)
-}
-
-func pow2(x int) int64 {
-	return int64(math.Pow(2, float64(x)))
-}
-
-// https://github.com/benbjohnson/testing
-// assert fails the test if the condition is false.
-func assert(t *testing.T, condition bool, msg string, v ...interface{}) {
-	t.Helper()
-
-	if !condition {
-		_, file, line, _ := runtime.Caller(1)
-		fmt.Printf("\033[31m%s:%d: "+msg+"\033[39m\n\n", append([]interface{}{filepath.Base(file), line}, v...)...)
-		t.FailNow()
-	}
-}
-
-// ok fails the test if an err is not nil.
-func ok(t *testing.T, err error) {
-	t.Helper()
-
-	if err != nil {
-		_, file, line, _ := runtime.Caller(1)
-		fmt.Printf("\033[31m%s:%d: unexpected error: %s\033[39m\n\n", filepath.Base(file), line, err.Error())
-		t.FailNow()
-	}
-}
-
-// nok fails the test if an err is nil.
-func nok(t *testing.T, err error) {
-	t.Helper()
-
-	if err == nil {
-		_, file, line, _ := runtime.Caller(1)
-		fmt.Printf("\033[31m%s:%d: expected error: %s\033[39m\n\n", filepath.Base(file), line, err.Error())
-		t.FailNow()
-	}
-}
+func TestDatasets(t *testing.T) {
+	defer setupZPool(t).cleanUp()
 
-// equals fails the test if exp is not equal to act.
-func equals(t *testing.T, exp, act interface{}) {
-	t.Helper()
+	_, err := zfs.Datasets("")
+	ok(t, err)
 
-	if !reflect.DeepEqual(exp, act) {
-		_, file, line, _ := runtime.Caller(1)
-		fmt.Printf("\033[31m%s:%d:\n\n\texp: %#v\n\n\tgot: %#v\033[39m\n\n", filepath.Base(file), line, exp, act)
-		t.FailNow()
+	ds, err := zfs.GetDataset("test")
+	ok(t, err)
+	equals(t, zfs.DatasetFilesystem, ds.Type)
+	equals(t, "", ds.Origin)
+	if runtime.GOOS != "solaris" {
```

Comment:
> I didn't review zfs_test.go very closely. I'm assuming that the tests are generally still the same...

Yep tests still the same, just dedented one tab for the most part (and a simple string test fail update).

Reply:
> I wonder if/why this is needed...

No clue, I'd like to try setting up a vagrant vm with omnios and smartos for some tests. iirc this may have come in from when someone from oracle PR'd some solaris stuff. But I don't have a way to test that so omni/smart os are going to be my solaris boxes for features and what not.

Reply:
OmniOS VM in GitHub actions:
https://github.com/nshalman/fsnotify/blob/fen-v2/.github/workflows/slow-tests.yml